gloo/test/cuda_base_test.h

```diff
@@ -36,17 +36,30 @@ class CudaFixture : public Fixture<T> {
     }
   }
 
+  // Upon returning, the memory allocated by this fixture is set. The
+  // fixture data is visible to any kernel launched on any stream.
   void assignValues() {
     Fixture<T>::assignValues();
     for (auto i = 0; i < cudaSrcs.size(); i++) {
-      CUDA_CHECK(cudaMemcpy(
+      CudaDeviceScope scope(cudaStreams[i].getDeviceID());
+      CUDA_CHECK(cudaMemcpyAsync(
         *cudaSrcs[i],
         this->srcs[i].get(),
         cudaSrcs[i].bytes,
-        cudaMemcpyHostToDevice));
+        cudaMemcpyHostToDevice,
+        *cudaStreams[i]));
+    }
+    // Synchronize every stream to ensure the memory copies have completed.
+    for (auto i = 0; i < cudaSrcs.size(); i++) {
+      CudaDeviceScope scope(cudaStreams[i].getDeviceID());
+      CUDA_CHECK(cudaStreamSynchronize(*cudaStreams[i]));
```

Comment:
Consider this for the sake of DRY:
```cpp
void assignValues() {
  assignValuesAsync();
  synchronizeCudaStreams();
}
```

Reply:
Yeah, looks nicer..

Reply:
But then we have the cudaSleep in the middle!

Reply:
Going with it as is